pkg/build/admission/overrides/admission.go

```diff
@@ -1,116 +1,93 @@
+		return BuildOverrides{}, errs.ToAggregate()
 	}
+	glog.V(4).Infof("Initialized build overrides plugin with config: %#v", *config)
+	return BuildOverrides{config: config}, nil
 }
 
-// Admit appplies configured overrides to a build in a build pod
-func (a *buildOverrides) Admit(attributes admission.Attributes) error {
-	if a.overridesConfig == nil {
-		return nil
-	}
-	if !buildadmission.IsBuildPod(attributes) {
+// ApplyOverrides applies configured overrides to a build in a build pod
+func (b BuildOverrides) ApplyOverrides(pod *kapi.Pod) error {
+	if b.config == nil {
 		return nil
 	}
-	return a.applyOverrides(attributes)
-}
 
-func (a *buildOverrides) applyOverrides(attributes admission.Attributes) error {
-	build, version, err := buildadmission.GetBuild(attributes)
+	build, version, err := buildadmission.GetBuildFromPod(pod)
 	if err != nil {
 		return err
 	}
-	glog.V(4).Infof("Handling build %s/%s", build.Namespace, build.Name)
 
-	if a.overridesConfig.ForcePull {
-		if err := applyForcePullToBuild(build, attributes); err != nil {
-			return err
+	glog.V(4).Infof("Applying overrides to build %s/%s", build.Namespace, build.Name)
+
+	if b.config.ForcePull {
+		if build.Spec.Strategy.DockerStrategy != nil {
+			glog.V(5).Infof("Setting docker strategy ForcePull to true in build %s/%s", build.Namespace, build.Name)
+			build.Spec.Strategy.DockerStrategy.ForcePull = true
+		}
+		if build.Spec.Strategy.SourceStrategy != nil {
+			glog.V(5).Infof("Setting source strategy ForcePull to true in build %s/%s", build.Namespace, build.Name)
+			build.Spec.Strategy.SourceStrategy.ForcePull = true
+		}
+		if build.Spec.Strategy.CustomStrategy != nil {
+			err := applyForcePullToPod(pod)
+			if err != nil {
+				return err
+			}
+			glog.V(5).Infof("Setting custom strategy ForcePull to true in build %s/%s", build.Namespace, build.Name)
+			build.Spec.Strategy.CustomStrategy.ForcePull = true
 		}
 	}
 
 	// Apply label overrides
-	for _, lbl := range a.overridesConfig.ImageLabels {
+	for _, lbl := range b.config.ImageLabels {
 		glog.V(5).Infof("Overriding image label %s=%s in build %s/%s", lbl.Name, lbl.Value, build.Namespace, build.Name)
 		overrideLabel(lbl, &build.Spec.Output.ImageLabels)
 	}
 
-	return buildadmission.SetBuild(attributes, build, version)
-}
-
-func applyForcePullToBuild(build *buildapi.Build, attributes admission.Attributes) error {
-	if build.Spec.Strategy.DockerStrategy != nil {
-		glog.V(5).Infof("Setting docker strategy ForcePull to true in build %s/%s", build.Namespace, build.Name)
-		build.Spec.Strategy.DockerStrategy.ForcePull = true
+	if len(b.config.NodeSelector) != 0 && pod.Spec.NodeSelector == nil {
+		pod.Spec.NodeSelector = map[string]string{}
 	}
-	if build.Spec.Strategy.SourceStrategy != nil {
-		glog.V(5).Infof("Setting source strategy ForcePull to true in build %s/%s", build.Namespace, build.Name)
-		build.Spec.Strategy.SourceStrategy.ForcePull = true
+	for k, v := range b.config.NodeSelector {
```

Comment:
do we want to do the same thing as the project nodeSelector and error if the build already specifies a conflicting key/value, or just stomp it?